### PR TITLE
verifying that this works against the latest wascc codec

### PR DESCRIPTION
**Pull request by autodidaddict**
This is a WIP because it is pointing to a local version of the new wascc-code (PR open).

Basically this just provides an implementation of the new `stop` function that's part of the new capability provider trait. It also derives clone on the provider which is now a requirement.

Also also - the basic types are now coming from the widl code generator and not from the shared data structures that used to be in wascc-codec.

### CODE REVIEW

**Comment by autodidaddict** [comment]
Obviously the build is going to fail because this PR still has a relative path reference to wascc-codec.

File: src/generated/core.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+use rmps::{Deserializer, Serializer};
+use serde::{Deserialize, Serialize};
+use std::io::Cursor;
+
+extern crate log;
+//extern crate wapc_guest as guest;
+//use guest::prelude::*;
+
+//use lazy_static::lazy_static;
+//use std::sync::RwLock;
```
Comment (brooksmtownsend) [inline src/generated/core.rs:11]: Are these leftover remnants, or are they generated as a comment?

File: src/generated/http.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    Host::default()
+}
+
+impl Host {}
+
+pub struct Handlers {}
+
+impl Handlers {}
+
+//lazy_static! {}
```
Comment (brooksmtownsend) [inline src/generated/http.rs:43]: Same as above

File: src/lib.rs [hunk 3]
```diff
@@ -41,10 +30,13 @@ const CAPABILITY_ID: &str = "wascc:http_server";
 const VERSION: &str = env!("CARGO_PKG_VERSION");
 const REVISION: u32 = 2; // Increment for each crates publish
 
+const OP_HANDLE_REQUEST: &str = "HandleRequest";
```
Comment (brooksmtownsend) [inline src/lib.rs:33]: Will an implementer of a capability contract need to define their own constants for the operations, or is this just a placeholder until we end up `codegen`ning them?

**Review by brooksmtownsend — COMMENTED** [review]
Largely looks good, a few comments

**Comment by autodidaddict** [comment]
The stuff that's commented out is what I commented out of the code generated output. As the generator gets better we can do less post processing. 

For now, it's up to the implementors to define the constants. We should be able to generate that as well at some point.

File: src/lib.rs (hunk 3 above)
Comment (autodidaddict) [reply src/lib.rs:33]: yeah, we'll codegen constants eventually from the `interface` portion of the widl.

**Pull request merged**